### PR TITLE
Add pljava_classpath_insecure GUC

### PR DESCRIPTION
**Pull request by davecramer**
In production pljava_classpath can only be set by super user or
in postgresql.conf
In order to allow developers to set it for testing we add a
GUC called pljava_classpath_insecure.
This can only be set in postgresql.conf or a super user
When it is true, pljava_classpath can be set by a user.

### CODE REVIEW

**Comment by ivannovick** [comment]
+1 for the general idea, this is helpful

File: src/backend/utils/misc/guc_gp.c [hunk 1]
```diff
@@ -3402,7 +3406,15 @@ struct config_bool ConfigureNamesBool_gp[] =
 		&vmem_process_interrupt,
 		false, NULL, NULL
 	},
-
+	{
+		{"pljava_classpath_insecure", PGC_POSTMASTER, CUSTOM_OPTIONS,
+			gettext_noop("Allow pljava_classpath to be set by user per session"),
+			NULL,
+			GUC_NO_SHOW_ALL | GUC_NOT_IN_SAMPLE
+		},
```
Comment (davecramer) [inline src/backend/utils/misc/guc_gp.c]: They can see it, they just can't see it with show all
show pljava_classpath_insecure will still work

Comment (ivannovick) [inline src/backend/utils/misc/guc_gp.c]: -1 for hiding it.  I think it should be a fully visible GUC

Comment (Eulerizeit) [inline src/backend/utils/misc/guc_gp.c]: I disagree. This is exposing a security hole.Why do we need to make it visible to everyone.

Comment (ivannovick) [inline src/backend/utils/misc/guc_gp.c]: The DBA is making an explicit decision to enable this, so they have opened the hole in order to improve productivity.  That is their decision.  By hiding gucs we just make our system more opaque.  In general I would recommend we don't have ANY hidden parameters in the product.

File: src/backend/utils/misc/guc_gp.c [hunk 2]
```diff
@@ -3402,7 +3406,15 @@ struct config_bool ConfigureNamesBool_gp[] =
 		&vmem_process_interrupt,
 		false, NULL, NULL
 	},
-
+	{
+		{"pljava_classpath_insecure", PGC_POSTMASTER, CUSTOM_OPTIONS,
+			gettext_noop("Allow pljava_classpath to be set by user per session"),
```
Comment (amilkh) [inline src/backend/utils/misc/guc_gp.c]: +1

File: src/backend/utils/misc/guc_gp.c [hunk 3]
```diff
@@ -5717,6 +5729,17 @@ assign_optimizer_log_failure(const char *val, bool assign, GucSource source)
...
+assign_pljava_classpath_insecure(bool newval, bool doit, GucSource source)
+{
+	if ( newval == true )
+	{
+		struct config_generic *pljava_cp = find_option("pljava_classpath", false, ERROR);
+		pljava_cp->context = PGC_USERSET;
+	}
+	return true;
+}
+
```
Comment (amilkh) [inline src/backend/utils/misc/guc_gp.c]: Looks like `pljava_classpatch` can only be set to true and setting it to false is silently ignored?
If that's intentional, might be good to add a code comment explaining why or printing a LOG message.

**Comment by amilkh** [comment]
Hmm, +1 as it can't seem to hurt anything.
http://gpdb.docs.pivotal.io/4380/guc_config-pljava_classpath.html

**Comment by pivotal-mike** [comment]
@davecramer are we OK to clean this up and push or does it need more work?

**Comment by davecramer** [comment]
Hmm I would have thought that was already in. Yes I will clean this up and push it

**Pull request merged**